src/System.Private.CoreLib/shared/System/Runtime/CompilerServices/AsyncMethodBuilder.cs

```diff
@@ -168,7 +164,11 @@ private void NotifySynchronizationContextOfCompletion()
         }
 
         /// <summary>Lazily instantiate the Task in a non-thread-safe manner.</summary>
-        private Task Task => _builder.Task;
+        public Task Task
+        {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
This is AggressivelyInlined but InitializeTaskAsPromise is NoInlined

> not a fan of gettors that have side-effects either. 

Its a Lazy initialize, rather than full side effect

Reply:
Would it be better to have `InitializeTaskAsPromise` inlined too?

Reply:
No; as isn't always called; the backing task can be set elsewhere (and I think usually is)